### PR TITLE
Auth 2304/hide unassigned if only one allowed

### PR DESCRIPTION
**Pull request by blakestier**
### :nut_and_bolt: Description: What code changed, and why?

Users were confused by the `(unassigned)` label in the dropdown when it was the only "project" to which they had access.

### :+1: Definition of Done

All these scenarios are true:

- [x] Hide the (unassigned) label from the top-right if that's the only thing a user has access to
- [x] If they have access to only 1 project, that projects name should still appear in the top-right
- [x] If a user has access to unassigned and another project, unassigned should still show up in the top-right if the user filters to view only unassigned resources

### :athletic_shoe: How to Build and Test the Change

Bring up the UI and sign in as users with varying project permissions (one project, one project + unassigned, only unassigned, no projects) and ensure that the above scenarios render correctly.

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### CODE REVIEW

File: components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.spec.ts [hunk 1]
```diff
@@ -239,9 +239,14 @@ describe('projectsFilterReducer', () => {
         genUnassignedProject()
       );
 
-      it('displays "(unassigned)"', () => {
+      it('does not display filter', () => {
+        const { filterVisible } = projectsFilterReducer(initialState, action);
+        expect(filterVisible).toEqual(false);
```
Comment (msorens) [inline components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.spec.ts]: Suggest also add a check for this to be true in at least one case that is **not** just unassigned.

File: components/automate-ui/src/app/page-components/projects-filter/projects-filter.component.html [hunk 2]
```diff
@@ -5,5 +5,6 @@
   [selectionCountVisible]="projectsFilter.selectionCountVisible$ | async"
   [selectionCountActive]="projectsFilter.selectionCountActive$ | async"
   [dropdownCaretVisible]="projectsFilter.dropdownCaretVisible$ | async"
+  [filterVisible]="projectsFilter.filterVisible$ | async"
```
Comment (msorens) [inline components/automate-ui/src/app/page-components/projects-filter/projects-filter.component.html]: I suspect you will need to set this in some unit tests for projects-filter-dropdown.component.ts, too.

File: components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.spec.ts (hunk 1 above)
Comment (blakestier) [reply components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.spec.ts]: I added it everywhere to be consistent with the testing suite. Feels a little redundant, but such is testing.

File: components/automate-ui/src/app/page-components/projects-filter/projects-filter.component.html (hunk 2 above)
Comment (blakestier) [reply components/automate-ui/src/app/page-components/projects-filter/projects-filter.component.html]: I did, thank you!

**Pull request merged**